src/libraries/HookLib.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {MinimalDelegationStorage, MinimalDelegationStorageLib} from "./MinimalDelegationStorage.sol";
+import {IHook} from "../interfaces/IHook.sol";
+
+enum HookFlags {
+    VERIFY_SIGNATURE, // 1 << 0
+    VALIDATE_USER_OP, // 1 << 1
+    IS_VALID_SIGNATURE, // 1 << 2
+    BEFORE_EXECUTE, // 1 << 3
+    AFTER_EXECUTE // 1 << 4
+
+}
+
+// HookId combines an address with flags: address << 160 | flags
+type HookId is uint256;
+
+library HookLib {
+    using MinimalDelegationStorageLib for MinimalDelegationStorage;
+
+    /// @dev Get the hook for a given keyHash and flag
+    function get(bytes32 keyHash, HookFlags flag) internal view returns (IHook) {
```

Comment:
i think this could use some library abstraction clean up but we can do it later

ie get should just return the registered validation module and THEN we should just if that validation module has the permission to be called out to

can do it as a todo tho

Reply:
#62